java/test/com/google/jglome/GlomeTest.java

```diff
@@ -0,0 +1,152 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.jglome;
+
+import static com.google.jglome.TestVector.TEST_VECTORS;
+import static java.security.spec.NamedParameterSpec.X25519;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.XECPrivateKey;
+import java.security.interfaces.XECPublicKey;
+import java.util.Arrays;
+import java.util.Optional;
+import com.google.jglome.Glome.WrongTagException;
+import org.junit.jupiter.api.Test;
+import com.google.jglome.Glome.CounterOutOfBoundsException;
+import com.google.jglome.Glome.MinPeerTagLengthOutOfBoundsException;
+import com.google.jglome.Glome.StaticSecureRandom;
+
+/**
+ * Contains tests for main.com.google.jglome.Glome functionality.
+ */
+public class GlomeTest {
+
+  final static int N_TEST_VECTORS = TEST_VECTORS.size();
+
+  Glome[][] glomeManagers = new Glome[N_TEST_VECTORS][2]; // first is for A, second is for B
+  KeyPair[] aKeys = new KeyPair[N_TEST_VECTORS];
+  KeyPair[] bKeys = new KeyPair[N_TEST_VECTORS];
+
+  GlomeTest() throws MinPeerTagLengthOutOfBoundsException, InvalidKeyException {
+    for (int i = 0; i < N_TEST_VECTORS; i++) {
+      aKeys[i] = deriveKeysFromPrivate(TEST_VECTORS.get(i).getKah());
+      bKeys[i] = deriveKeysFromPrivate(TEST_VECTORS.get(i).getKbh());
+      glomeManagers[i][0] = new Glome((XECPublicKey) bKeys[i].getPublic(),
+          (XECPrivateKey) aKeys[i].getPrivate(), 32);
+      glomeManagers[i][1] = new Glome((XECPublicKey) aKeys[i].getPublic(),
+          (XECPrivateKey) bKeys[i].getPrivate(), 28);
+    }
+  }
+
+  @Test
+  public void testShouldFailWhenMinPeerTagLengthIsOutOfBounds() throws InvalidKeyException {
+    int[] minPeerTagLength = new int[]{0, 33};
+
+    for (int len : minPeerTagLength) {
+      try {
+        new Glome((XECPublicKey) aKeys[0].getPublic(), (XECPrivateKey) bKeys[0].getPrivate(), len);
+      } catch (MinPeerTagLengthOutOfBoundsException e) {
+        assertEquals(e.getMessage(),
+            String.format("minPeerTagLength argument should be in [1..32] range. Got %d.", len));
+      }
+    }
+  }
+
+  @Test
+  public void testShouldFailWhenCounterIsOutOfBounds() {
```

Comment:
This should be two tests: one for out of bounds and if you really want one that iterates over all of the possible counters (although that is somewhat unusual). Like this you do ensure that the message matches but you don't actually check the expected case is only hit for -1 and 256.